File: versions/1.8.9-forge/src/main/java/app/qwertz/quickjoin/QuickJoin.java
package app.qwertz.quickjoin;

import app.qwertz.quickjoin.config.QuickJoinConfig;
import app.qwertz.quickjoin.command.QuickJoinCommand;
import app.qwertz.quickjoin.favorites.FavoritesManager;
import net.minecraft.client.Minecraft;
import net.minecraftforge.client.ClientCommandHandler;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;

@Mod(modid = QuickJoin.MODID, name = QuickJoin.NAME, version = QuickJoin.VERSION)
public class QuickJoin {

    public static QuickJoinConfig config;
    public boolean isEnabled() {
        return QuickJoin.config.enabled;
    }
    @Mod.EventHandler
    public void onInit(FMLInitializationEvent event) {
        config = QuickJoinConfig.INSTANCE;
        MinecraftForge.EVENT_BUS.register(new CommandEventHandler(this));
        ClientCommandHandler.instance.registerCommand(new QuickJoinCommand(this));
        FavoritesManager.initialize(new java.io.File(Minecraft.getMinecraft().mcDataDir, "config").toPath());
    }

    public static final String MODID = "@ID@";
    public static final String NAME = "@NAME@";
    public static final String VERSION = "@VER@";

    @Mod.Instance(MODID)
    public static QuickJoin INSTANCE;
}

File: src/main/kotlin/app/qwertz/quickjoin/QuickJoin.kt
package app.qwertz.quickjoin

import app.qwertz.quickjoin.QuickJoin.Companion.config
import app.qwertz.quickjoin.command.IsEnabled
import app.qwertz.quickjoin.command.QuickJoinCommand
import app.qwertz.quickjoin.gui.QuickJoinGui
import cc.polyfrost.oneconfig.utils.gui.GuiUtils
import cc.polyfrost.oneconfig.utils.hypixel.HypixelUtils
import net.minecraft.client.Minecraft
import net.minecraft.util.ChatComponentText
import net.minecraftforge.client.ClientCommandHandler
import net.minecraftforge.client.event.ClientChatReceivedEvent
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@Mod(modid = QuickJoin.MODID, name = QuickJoin.NAME, version = QuickJoin.VERSION)
class QuickJoin {
    // Register the config and commands.

    @Mod.EventHandler
    fun onInit(event: FMLInitializationEvent?) {
        config = app.qwertz.quickjoin.config.QuickJoinConfig
        MinecraftForge.EVENT_BUS.register(CommandEventHandler())
        ClientCommandHandler.instance.registerCommand(QuickJoinCommand())
    }
    companion object {
        const val MODID: String = "@ID@"
        const val NAME: String = "@NAME@"
        const val VERSION: String = "@VER@"

        @Mod.Instance(MODID)
        lateinit var INSTANCE: QuickJoin
        lateinit var config: app.qwertz.quickjoin.config.QuickJoinConfig
    }
}



class CommandEventHandler {
    @SubscribeEvent
    fun onChatReceived(event: ClientChatReceivedEvent) {
        val message = event.message.unformattedText
        if (message.startsWith("Unknown command.") && config.EnableAlias) {
            val pattern = Regex("""'([^']*)'""")
            val matchResult = pattern.find(message)?.value
            val alias = config.CommandAlias
            if (matchResult == "'$alias'") {
                if (IsEnabled().enabledCheck()) {
                    GuiUtils.displayScreen(QuickJoinGui())
                } else {
                    if (HypixelUtils().isHypixel) {
                        if (config.DebugMode) {
                            Minecraft.getMinecraft().thePlayer.addChatMessage(ChatComponentText("§4[§6§lQUICKJOIN-DEBUG§4]§a: Displaying MainGui"))
                        }
                        GuiUtils.displayScreen(QuickJoinGui())
                    }
                    else {
                        Minecraft.getMinecraft().thePlayer.addChatMessage(ChatComponentText("§4[§6§lQUICKJOIN§4]§a: You are not on Hypixel."))
                    }
                }

            }
        }
    }
}